### PR TITLE
KFSPTS-13246: Implementation of new button that allows user to delete a line from Applied Funds section of CGB PaymentApplication.

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

**Comment by nkimble** [comment]
retest this please

**Comment by nkimble** [comment]
retest this please

**Comment by nkimble** [comment]
KualiCo. is making numerous changes to this module that will be release soon.
This code in targeted to become a contribution to base code. 
Our changes could not be made as institutional changes and placed on our package branch because of the way the struts config wild card loads the "ar" module.

Due to all of these issues, the original methods in the class files requiring changes were retained with the modified method following to make it easier for the developer to who will provide our contribution back to KualiCo.

**Comment by nkimble** [comment]
ChadH: 
Please let me know if you need this PR closed and recreated so that these changes can be rebased to prevent possible merge conflicts that would result later on.

File: src/main/webapp/WEB-INF/tags/module/ar/paymentApplicationSummaryOfAppliedFunds.tag [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+												</td>
+												<td>
+													<c:out value="${invoicePaidApplied.invoiceDetail.invoiceItemDescription}" />&nbsp;
+												</td>
+												<td style="text-align: right;">
+													$<c:out value="${invoicePaidApplied.invoiceItemAppliedAmount}" />
+												</td>
+												<%-- Cornell mod is addition of this td tag --%>
+												<td>
+												    <html:html-button
```
Comment (cah292) [inline src/main/webapp/WEB-INF/tags/module/ar/paymentApplicationSummaryOfAppliedFunds.tag]: You should wrap this in an "if" block that checks the status of the "readOnly" attribute, to ensure that this button is hidden when the document is in a non-editable state. There should be some example areas in this file that check the "readOnly" attribute's value.

File: src/main/java/org/kuali/kfs/module/ar/dataaccess/impl/InvoicePaidAppliedDaoObj.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.ojb.broker.query.Criteria;
+import org.apache.ojb.broker.query.QueryByCriteria;
+import org.apache.ojb.broker.query.QueryFactory;
+import org.kuali.kfs.module.ar.businessobject.InvoicePaidApplied;
+import org.kuali.rice.core.framework.persistence.ojb.dao.PlatformAwareDaoBaseOjb;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.springframework.transaction.annotation.Transactional;
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/org/kuali/kfs/module/ar/dataaccess/impl/InvoicePaidAppliedDaoObj.java]: Removing.

File: src/main/java/org/kuali/kfs/module/ar/dataaccess/impl/InvoicePaidAppliedDaoObj.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.kuali.kfs.module.ar.dataaccess.InvoicePaidAppliedDao;
+
+public class InvoicePaidAppliedDaoObj extends PlatformAwareDaoBaseOjb implements InvoicePaidAppliedDao {
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/org/kuali/kfs/module/ar/dataaccess/impl/InvoicePaidAppliedDaoObj.java]: Fixed.

File: src/main/java/org/kuali/kfs/module/ar/dataaccess/impl/InvoicePaidAppliedDaoObj.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    private static final Logger LOG = LogManager.getLogger(InvoicePaidAppliedDaoObj.class);
+    
+    private static final String DOCUMENT_NUMBER = "documentNumber";
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/org/kuali/kfs/module/ar/dataaccess/impl/InvoicePaidAppliedDaoObj.java]: Fixed.

File: src/main/webapp/WEB-INF/tags/module/ar/paymentApplicationSummaryOfAppliedFunds.tag (hunk 1 above)
Comment (nkimble) [reply src/main/webapp/WEB-INF/tags/module/ar/paymentApplicationSummaryOfAppliedFunds.tag]: Code adjusted.

File: src/main/java/org/kuali/kfs/module/ar/document/PaymentApplicationDocument.java [hunk 5]
```diff
@@ -0,0 +1,1566 @@
...
+    }
+
+    /**
+     * Collects and returns the combined distributions from NonInvoiced/NonAr and Unapplied. This method is intended to be used only
+     * when the document has gone to final, to show what control documents were issued what funds. The return value is a
+     * Map<String,KualiDecimal> where the key is the NonAppliedHolding's ReferenceFinancialDocumentNumber and the value is the
+     * Amount to be applied.
+     *
+     * @return
+
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/org/kuali/kfs/module/ar/document/PaymentApplicationDocument.java]: The data List this flag is tracking the changes for is declared in the PaymentApplicationDocument (List<InvoicePaidApplied> invoicePaidApplieds). That is why I put the flag in this document class and not on the form.  
Do think that we really want to move it? 
Shouldn't they be together in the same class?

**Comment by nkimble** [comment]
I believe that I locally tested the scenario you described but I want to run through it again after the all the code review changes are in in place.  I will report my findings after we come to a consensus on that one open issue and I have performed local validation again.

File: src/main/java/org/kuali/kfs/module/ar/document/PaymentApplicationDocument.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/java/org/kuali/kfs/module/ar/document/PaymentApplicationDocument.java]: In that case, I think it's fine to keep the flag on the document. Thanks for the explanation.

**Comment by nkimble** [comment]
Closed this PR due to merge conflicts attempting to add code review changes. A new PR will be created for this change.

**Pull request closed without merging**